torchgeo/transforms/temporal.py

```diff
@@ -68,3 +71,29 @@ def compute_transformation(
         """
         out = self.identity_matrix(input)
         return out
+
+
+class TemporalEmbedding(nn.Module):
+    """Generic sinusoidal embedding for periodic temporal features."""
+
+    def __init__(self, period: int) -> None:
+        """Initialize a TemporalEmbedding instance.
+
+        Args:
+            period: The period of the sinusoidal function.
+        """
+        super().__init__()
+        self.period = period
+
+    def forward(self, t: Tensor) -> Tensor:
+        """Compute sinusoidal embeddings for periodic time values.
+
+        Args:
+            t: Tensor of shape (B,) or (B, 1), representing time values.
```

Comment:
I wonder if this transform would only be useful if it took in a sample as input and then used the `time` key (or a user-defined key). Currently this transform requires custom handling.